### PR TITLE
Revert the revert: Practice for code reviews

### PR DESCRIPTION
**Pull request by cozos**
This reverts commit cc168e8cf03b230cf19f6ac642a70a9f8fbcc31e.

### CODE REVIEW

File: src/main/HelloWorldMiningCompany.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public void onStart() {
+        game = mirror.getGame();
+        self = game.self();
+        
+        unitDepartment = new UnitDepartment(game, self);
+        unitDepartment.recruit();
+        
+        buildingDepartment = new BuildingDepartment(game, self);
+        buildingDepartment.recruit();
+
```
Comment (cozos) [inline src/main/HelloWorldMiningCompany.java:37]: This line is bad because: _____

Comment (cozos) [inline src/main/HelloWorldMiningCompany.java]: @anava P.S. Better to comment on the file lines.

File: src/departments/IDepartment.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+public interface IDepartment {
+  public enum Purpose{
+    GAMESTART,
+    MINE,
+    SCOUT,
+    BUILD,
+    DEFEND,
+    ATTACK,
+    CRITICAL
```
Comment (anava) [inline src/departments/IDepartment.java]: What's a critical purpose?

File: src/departments/IDepartment.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  }
+  
+  public void init();
+  public List<Unit> getUnits();
+  public void requestFunds(Team team, Purpose purpose,int minerals, int vespene, int Suggestion);
```
Comment (anava) [inline src/departments/IDepartment.java:12]: Why not use the Request object instead of having a lot of params?

Comment (cozos) [inline src/departments/IDepartment.java:12]: Todo

File: src/departments/IDepartment.java (hunk 2 above)
Comment (cozos) [inline src/departments/IDepartment.java]: Presumably, something that is critical to the victory condition and is given the highest priority i.e. pull workers from mining to defend against zerg rush, or sending all units to enemy base guarantees victory

TODO: Also, the purpose enum should probably be in the Request class.

**Pull request merged**